daff/README.md

```diff
@@ -0,0 +1,110 @@
+# Directory Contents
+
+**diff2/** - This folder contains html diff2 table output of `diff2.js` script.
+
+**diff3/** - This folder contains html diff3 table output of `diff3.js` script.
+
+**mergedData/** - This folder contains updated local file after having diff changes merged in. It is an output of `diff3.js`.
+
+**data/** - This folder contains input files for diff scripts, metadata of input files and results of runinng test cases.
+
+**calcColumnSimilarity.js** - This script completes preproccessing steps for `diff2.js`/`diff3.js` and produces meta data
+for its given input data files.
+
+How to run
+
+* For diff2:
+```node calcColumnSimilarity.js local.csv remote.csv```
+
+* For diff3:
+```node calcColumnSimilarity.js ancestor.csv local.csv remote.csv```
+
+* For diff3 with custom column similarity threshold:
+```node calcColumnSimilarity.js ancestor.csv local.csv remote.csv 0.5```
+
+Output file
+```node data/meta_local_remote.json```
+
+**diff2.js** - This script runs diff2 for two given input data files and outputs a visualization of the diff in an html file.
+
+How to run
+```node diff2.js local.csv remote.csv```
+
+Output file
+```diff2/diff2_local_remote.html```
+
+**diff3.js** - This script runs diff3 for three given input data files and outputs a visualization of the diff
+in an html file and a updated local file after having diff changes merged in.
+
+How to run
+```node diff3.js local.csv remote.csv```
+
+Output file
+```mergedData/merged_ancestor_local_remote.html```
+```mergedData/merged_ancestor_local_remote.html```
+
+## Test Cases
+
+All test case directories contain the same input data files:
+
+* `ancestor.csv`
+* `local.csv`
+* `remote.csv`
+
+All test cases generate same set of output files:
+
+* `diff2_local_remote.html`
+* `diff3_ancestor_local_remote.html`
+
+### Case 1
+
+Test diff2 and diff3 without any preprocessing and postprocessing. All 3 files have their
+own columns/rows and common columns/rows.
+
+### Case 2
+
+This test considers the case where preprocessing removed columns in ancestor and
```

Comment:
For cases where the preprocessing is assumed, it might be better to add one more point of `Prerequisite: ...` section in the description. For example:

`Prerequisite: Assume that the preprocessing has performed. This preprocessing removes columns that the local file doesn’t care about from the ancestor and remote files.`